docs/src/references/c8y-configuration-management.md

```diff
@@ -0,0 +1,162 @@
+# Device Configuration Management using Cumulocity
+
+Thin-edge provides an operation plugin to
+[manage device configurations using Cumulocity](https://cumulocity.com/guides/users-guide/device-management/#to-retrieve-and-apply-a-configuration-snapshot-to-a-device-which-supports-multiple-configuration-types).
+
+* This management is bi-directional:
+  * A device can be taken as reference, all the managed files being uploaded to the cloud
+    and stored there as a configuration snapshot.
+  * A configuration snapshot can be pushed from the cloud to any devices of the same type,
+    i.e. supporting the same kind of configuration files.
+* With this operation plugin, the device owner defines the list of files
+  (usually configuration files, but not necessarily),
+  that will be managed from the cloud tenant.
+* Notably, the plugin configuration itself is managed from the cloud,
+  meaning, the device owner can update from the cloud the list of files to be managed.
+* By default, the managed files are represented on the cloud by their full path on the device.
+  Optionally, the device owner can assign an alias to each target path.  
+* When files are downloaded from the cloud to the device,
+  these files are stored in a temporary directory first.
+  They are atomically moved to their target path, only after a fully successful download.
+  The aim is to avoid breaking the system with half downloaded files.
+* Once all the files of a snapshot have been downloaded,
+  the plugin published a notification message on the local thin-edge MQTT bus.
+  The device software has to subscribe to these messages if any action is required,
+  say to check the content of file, to preprocess it or to restart a daemon. 
+* In other words, the responsibilities of the plugin are:
+  * to define the list of files under configuration management
+  * to notify the cloud when this list is updated,
+  * to upload these files to the cloud on demand,  
+  * to download the files pushed from the cloud,
+  * to make sure that the target files are updated atomically after successful download,
+  * to notify the device software when the configuration is updated.
+* By contrast, the plugin is not responsible for:
+  * checking the uploaded files are well-formed,
+  * restarting the configured processes.
+* A user-specific component, installed on the device,
+  can implement more sophisticated configuration use-cases by:
+  * listening for configuration updates on the local thin-edge MQTT bus,
+  * restarting the appropriate processes when appropriate,  
+  * declaring intermediate files as the managed files,
+    to have the opportunity to check or update their content
+    before moving them to the actual targets.
+
```

Comment:
How many previous versions of the `config files` will be stored on the device?.

Reply:
None.